src/components/dashboard/Dashboard.test.jsx

```diff
@@ -0,0 +1,130 @@
+import { CourseContextProvider } from '../course/CourseContextProvider';
+import { COURSE_PACING_MAP } from '../course/data/constants';
+import { TEST_OWNER } from '../course/tests/data/constants';
+
+import {
+  renderWithRouter, fakeReduxStore,
+} from '../../utils/tests';
+import Dashboard, { LICENCE_ACTIVATION_MESSAGE } from './Dashboard';
+
+const mockStore = configureMockStore([thunk]);
+
+/* eslint-disable react/prop-types */
+const DashboardWithContext = ({
+  initialAppState = {},
+  initialUserSubsidyState = {},
+  initialCourseState = {},
+  initialReduxStore = fakeReduxStore,
+}) => (
+  <AppContext.Provider value={initialAppState}>
+    <UserSubsidyContext.Provider value={initialUserSubsidyState}>
+      <CourseContextProvider initialState={initialCourseState}>
+        <ReduxProvider store={mockStore(initialReduxStore)}>
+          <Dashboard />
+        </ReduxProvider>
+      </CourseContextProvider>
+    </UserSubsidyContext.Provider>
+  </AppContext.Provider>
+);
+/* eslint-enable react/prop-types */
+
+let mockLocation = {
+  pathname: '/welcome',
+  hash: '',
+  search: '',
+  state: { activationSuccess: true },
+};
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useLocation: () => (mockLocation),
+}));
+
+jest.mock('./sidebar/offers', () => ({
+  ...jest.requireActual('./sidebar/offers'),
+  fetchOffers: () => ({ type: 'fetch' }),
+}));
+
+describe('<Dashboard />', () => {
+  const initialAppState = { enterpriseConfig: { name: 'BearsRUs' } };
+  const initialUserSubsidyState = { hasAccessToPortal: true };
+  const mockWindowConfig = {
+    type: 'screen',
+    width: breakpoints.large.minWidth + 1,
+    height: 800,
+  };
+  const initialCourseState = {
+    course: {
+      subjects: [{
+        name: 'Test Subject 1',
+        slug: 'test-subject-slug',
+      }],
+      shortDescription: 'Course short description.',
+      title: 'Test Course Title',
+      owners: [TEST_OWNER],
+      programs: [],
+      image: {
+        src: 'http://test-image.url',
+      },
+    },
+    activeCourseRun: {
+      isEnrollable: true,
+      key: 'test-course-run-key',
+      pacingType: COURSE_PACING_MAP.SELF_PACED,
+      start: '2020-09-09T04:00:00Z',
+      availability: 'Current',
+      courseUuid: 'Foo',
+    },
+    userEnrollments: [],
+    userEntitlements: [],
+    catalog: {
+      containsContentItems: true,
+    },
+  };
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('renders license activation alert on activation success', () => {
+    renderWithRouter(
+      <DashboardWithContext initialAppState={initialAppState} initialUserSubsidyState={initialUserSubsidyState} />,
+      { route: '/?activationSuccess=true' },
+    );
+    expect(screen.getByText(LICENCE_ACTIVATION_MESSAGE)).toBeTruthy();
+  });
+  it('does not render license activation alert without activation success', () => {
+    // NOTE: This modifies the original mockLocation
```

Comment:
This is legit terrible and I tried about 50 different ways to mock it without doing this, and this was the only thing that worked. I'm *very* open to alternate mocking suggestions here.

Reply:
Hmmm yeah I've never tested anything that involved mocking/modifying the actual paths. Usually those would be separate components.

Thus I guess the only thought/question here is: are we testing at too high of a level? ie should we test at a more functional level rather than requiring routes here or is that just not possible for what we want to test?

Reply:
Yeah, that's not possible for what I'm testing here, because there's a popup based on what the location provider is giving me. I tried various ways of passing it into the router and hoping it did the right thing, but no dice.